### ISSUE TITLE
Rules for Memory appear to be wrong

### ISSUE THREAD

**Issue by bencyoung** [dotnet/docs#11411]
Rule #3 and #4 appear to be wrong where they place

```csharp
string defensiveCopy = message.ToString();
```

I think these should be called before the task is run (outside the delegate), else the copy can still be corrupted by the time it is made...

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 4819512e-1044-1c57-0de1-447164220fd5
* Version Independent ID: d6cf8d26-fb2f-7138-f0c7-ab529a5b9cde
* Content: [Memory&lt;T&gt; and Span&lt;T&gt; usage guidelines](https://docs.microsoft.com/en-us/dotnet/standard/memory-and-spans/memory-t-usage-guidelines#feedback)
* Content Source: [docs/standard/memory-and-spans/memory-t-usage-guidelines.md](https://github.com/dotnet/docs/blob/master/docs/standard/memory-and-spans/memory-t-usage-guidelines.md)
* Product: **dotnet**
* GitHub Login: @rpetrusha
* Microsoft Alias: **ronpet**

**Comment by kylejrp** [comment]
Discussion on Reddit: https://www.reddit.com/r/csharp/comments/b4np6w/memoryt_and_spant_usage_guidelines/ej85l5a?utm_source=share&utm_medium=web2x

**Comment by rpetrusha** [comment]
Thanks for your feedback, @bencyoung, and for the link to the discussion on Reddit, @kylejrp. I'll address these issues in our May sprint.

**Comment by rpetrusha** [comment]
This has been partially addressed in dotnet/samples#842.

**Comment by amosonn** [comment]
The example for #4 is still broken. For one, it doesn't return `Task`, even though the section discusses methods returning `Task`. This can be of course remedied by returning the `Task.Run`, but then we go back to an earlier example, and the defensive copy becomes unnecessary, missing the point.

Something like:
```
    static Task Log(ReadOnlyMemory<char> message)
    {
        // Run in the background so that we don't block the main thread while performing IO.
        Task<string> t = Task.Run(() => {
            string defensiveCopy = message.ToString();
            return defensiveCopy;
        }
        t.ContinueWith((defensiveCopyTask) => {
            string defensiveCopy = defensiveCopyTask.Result;
            StreamWriter sw = File.AppendText(@".\input-numbers.dat");
            sw.WriteLine(defensiveCopy);
            sw.Flush();
        });
        return (async () => await t)();
    }
```
Demonstrates the point, but is ugly and contrived.
